onboarding/source/amb_light_service.c

```diff
@@ -18,22 +28,71 @@ static void lightServiceTask(void * pvParameters);
 uint8_t initLightService(void) {
     /* USER CODE BEGIN */
     // Create the task and queue here.
+    BaseType_t xReturned = pdFAIL;
+    if (lightServiceHandle == NULL) {
+        // Create controller task
+        xReturned = xTaskCreate(lightServiceTask,             /* Function that implements the task. */
+                                LIGHT_SERVICE_NAME,            /* Text name for the task. */
+                                LIGHT_SERVICE_STACK_SIZE,      /* Stack size in words, not bytes. */
+                                NULL,                       /* Parameter passed into the task. */
+                                LIGHT_SERVICE_PRIORITY,        /* Priority at which the task is created. */
+                                &lightServiceHandle);     /* Used to pass out the created task's handle. */
+    }
+
+    // create queue
+    if(lightServiceQueue == NULL){
+        lightServiceQueue = xQueueCreate(LIGHT_SERVICE_QUEUE_LENGTH, LIGHT_SERVICE_ITEM_SIZE);
+    }
+    
+    // inititalize printing
+    sciMutexInit();
+
+    // initialize ADC 
+    adcInit();
 
     /* USER CODE END */
-    return 1;
+    return xReturned;
 }
 
 static void lightServiceTask(void * pvParameters) {
     /* USER CODE BEGIN */
     // Wait for MEASURE_LIGHT event in the queue and then print the ambient light value to the serial port.
+    while(1){
+        light_event_t *nextEvent = NULL;
```

Comment:
xQueueReceive expects a pointer to a buffer to store the data. This just declares a pointer; memory for the actual value hasn't been allocated.